client/extensions/woocommerce/app/promotions/promotion-models.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * External dependencies
+ */
+import { translate as moduleTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import CurrencyField from './fields/currency-field';
+import PercentField from './fields/percent-field';
+import TextField from './fields/text-field';
+
+export function createModels( translate ) {
+	// -------------------------
+	// Reusable Promotion Fields
+	// -------------------------
+	const couponCode = {
+		component: TextField,
+		labelText: translate( 'Coupon code' ),
+		explanationText: translate(
+			'Only apply this promotion when the customer supplies the coupon code'
+		),
+		placeholderText: translate( 'Enter coupon code' ),
+		isRequired: true,
+	};
+
+	const fixedDiscount = {
+		component: CurrencyField,
+		isRequired: true,
+	};
+
+	// ---------------
+	// Promotion types
```

Comment:
Maybe add a note that these are lists of fields, to be clear that's what a "model" is.

Reply:
I'm going to be adding to this structure a little bit, so it will change. But I'll keep this in mind and look for your feedback as I go. Thanks!

Reply:
I think with them having field in the names, and the JSDocs, this should be very evident now. Thanks!